contracts/0.8.9/oracle/BaseOracle.sol

```diff
@@ -0,0 +1,342 @@
+
+    /// @dev Storage slot: ConsensusReport consensusReport
+    bytes32 internal constant CONSENSUS_REPORT_POSITION =
+        keccak256("lido.BaseOracle.consensusReport");
+
+
+    uint256 public immutable SECONDS_PER_SLOT;
+    uint256 public immutable GENESIS_TIME;
+
+    ///
+    /// Initialization & admin functions
+    ///
+
+    constructor(uint256 secondsPerSlot, uint256 genesisTime) {
+        SECONDS_PER_SLOT = secondsPerSlot;
+        GENESIS_TIME = genesisTime;
+    }
+
+    /// @notice Returns the address of the HashConsensus contract.
+    ///
+    function getConsensusContract() external view returns (address) {
+        return CONSENSUS_CONTRACT_POSITION.getStorageAddress();
+    }
+
+    /// @notice Sets the address of the HashConsensus contract.
+    ///
+    function setConsensusContract(address addr) external onlyRole(MANAGE_CONSENSUS_CONTRACT_ROLE) {
+        _setConsensusContract(addr, LAST_PROCESSING_REF_SLOT_POSITION.getStorageUint256());
+    }
+
+    /// @notice Returns the current consensus version expected by the oracle contract.
+    ///
+    /// Consensus version must change every time consensus rules change, meaning that
+    /// an oracle looking at the same reference slot would calculate a different hash.
+    ///
+    function getConsensusVersion() external view returns (uint256) {
+        return CONSENSUS_VERSION_POSITION.getStorageUint256();
+    }
+
+    /// @notice Sets the consensus version expected by the oracle contract.
+    ///
+    function setConsensusVersion(uint256 version) external onlyRole(MANAGE_CONSENSUS_VERSION_ROLE) {
+        _setConsensusVersion(version);
+    }
+
+    ///
+    /// Data provider interface
+    ///
+
+    /// @notice Returns the last consensus report hash and metadata.
+    ///
+    function getConsensusReport() external view returns (
+        bytes32 hash,
+        uint256 refSlot,
+        uint256 receptionTime,
+        uint256 deadlineTime,
+        bool processingStarted
+    ) {
+        ConsensusReport memory report = _storageConsensusReport().value;
+        uint256 processingRefSlot = LAST_PROCESSING_REF_SLOT_POSITION.getStorageUint256();
+        return (
+            report.hash,
+            report.refSlot,
+            report.receptionTime,
+            report.deadlineTime,
+            report.refSlot != 0 && report.refSlot == processingRefSlot
+        );
+    }
+
+    ///
+    /// Consensus contract interface
+    ///
+
+    /// @notice Called by HashConsensus contract to push a consensus report for processing.
+    ///
+    /// Note that submitting the report doesn't require the oracle to start processing it
+    /// right away, this can happen later. Until the processing is started, HashConsensus is
+    /// free to reach consensus on another report for the same reporting frame and submit it
+    /// using this same function.
+    ///
+    function submitReport(bytes32 reportHash, uint256 refSlot, uint256 deadline) external {
```

Comment:
Let's rename the method to `submitConsensusReport` and add `submitReport` proxy which can be used by oracle daemons

Reply:
And maybe add `submitReport` proxy which can be used by oracle daemons